sami2py/tests/test_core_class.py

```diff
@@ -26,55 +29,80 @@ def teardown(self):
                 archive_file.write('')
                 sami2py.archive_dir = ''
 
-    @raises(IOError)
     def test_model_input_exception(self):
         """File not found error should be produced if the file does not exist
         """
-        sami2py.Model(tag='none', lon=428, day=428, year=1969)
+        with pytest.raises(IOError):
+            sami2py.Model(tag='none', lon=428, day=428, year=1969)
 
     def test_model_instantiation(self):
         """Test that model object is instantiated as a sami2py_model
         """
-        model = sami2py.Model(tag='test', lon=256, year=1999, day=256,
-                              test=True)
-        assert isinstance(model, sami2py.Model)
-
-    def test_model_instantiation_with_unformatted_files_and_mods(self):
-        """Test that model object is instantiated as a sami2py_model
-        """
-        model = sami2py.Model(tag='test', lon=256, year=1999, day=257,
-                              test=True)
+        model = sami2py.Model(tag='test', lon=self.lon, year=self.year,
+                              day=self.day, test=True, outn=True)
         assert isinstance(model, sami2py.Model)
 
     def test_model_plot(self):
         """Basic test that a reasonable plot has been created by testing the
            resulting axis limits
         """
         import matplotlib
-        matplotlib.use('Agg')
-        import matplotlib.pyplot as plt
-        model = sami2py.Model(tag='test', lon=256, year=1999, day=256,
-                              test=True)
-        model.plot_lat_alt()
-        fig = plt.gcf()
+
+        model = sami2py.Model(tag='test', lon=self.lon, year=self.year,
+                              day=self.day, test=True)
+        fig = model.plot_lat_alt()
+        assert isinstance(fig, matplotlib.figure.Figure)
+
+        tol = 1.e-4
         xlims = fig.axes[0].get_xlim()
         ylims = fig.axes[0].get_ylim()
-        assert xlims == (-36.3329, 19.37387)
-        assert ylims == (84.98926, 1999.998)
-        plt.close()
+        assert abs(xlims[0] + 36.3329) < tol
+        assert abs(xlims[1] - 19.37387) < tol
+        assert abs(ylims[0] - 84.98926) < tol
+        assert abs(ylims[1] - 1999.998) < tol

```

Comment:
Since this test is now run twice and the two data sets are slightly different, updates to use a tolerance rather than exact precision.